scripts/generate-test-binaries.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+indir=$1
+outdir=$2
+binary_name=$3
+
+for os in "darwin" "linux" "windows"; do
+  for arch in "amd64" "arm64"; do
+    echo "Building test binary for $os/$arch"
+    GOOS=$os GOARCH=$arch go build -o "$outdir/$binary_name-$os-$arch" "$indir"
```

Comment:
I think you should try to make these binaries reproducible so that if I run this script on my machine, it will generate the exact same files. You will at least `-trimpath` and `CGO_ENABLED=0`.

Alternatively, don't check in the binaries at all? Maybe we can make a `testing.Run` to generate them for the tests that need them. 

Anyway, this is all non-blocking, but I might try to fix it myself if I encounter problems.

Reply:
yeah, i think you're right actually, we should probably generate these on the fly like we do with the integration test. it might be a little slow, but tests are pretty fast as is.

Reply:
this is done :)

Reply:
Sorry I bikeshedded you into this,

```


C:\buildkite-agent\builds\agent-runners-windows-amd64-i-050772c7c334529b8-1\buildkite\agent\test\fixtures\hook\test-binary-windows-arm64
--



    type_test.go:98: Failed to remove test-binary-hook: remove 
C:\buildkite-agent\builds\agent-runners-windows-amd64-i-050772c7c334529b8-1\buildkite\agent\test\fixtures\hook\test-binary-linux-amd64:
 The process cannot access the file because it is being used by another 
process.
```

Maybe this just skip the removal on Windows, it should be cleaned out at the next checkout anyway?

It's a bit concerning that it says "cannot access". If that prevents executing the hook concurrently, we may have a problem. But I suspect it's just because we are trying to delete it and there a process hanging onto the binary.

Reply:
alternatively, i think i can just yeet them into the tempdir and let the os sort it out